response.go

```diff
@@ -48,9 +48,21 @@ func (r *Response) Encode(v interface{}) {
 	r.Header.Set("Content-Type", "application/json")
 }
 
+// MaskDownstreamErrors is a context key that can be used to enable
+// wrapping of downstream response errors on a per-request basis.
+//
+// This is implemented as a context key to allow us to migrate individual
+// services from the old behaviour to the new behaviour without adding a
+// dependency on config to Typhon.
+type MaskDownstreamErrors struct{}
```

Comment:
Not sure about `Mask` here but I thought it was the most descriptive. Alternatively `Wrap...`?

Reply:
Mask has a bit of a negative connotation for me so I'd personally go with Wrap, but I don't think it matters too much

Reply:
Yep I'll go with Wrap